lib/team_collection.rb

```diff
@@ -0,0 +1,18 @@
+require 'csv'
+require './lib/team'
+
+class TeamCollection
+  attr_reader :teams
+
+  def initialize(teams_data_input)
+    @teams_csv = teams_data_input
+    @teams = []
+  end
+
+  def create_teams
+    csv = CSV.read(@teams_csv, headers: true, converters: :numeric, header_converters: :symbol)
+    csv.map do |row|
+      @teams << Team.new(row.to_h)
+    end
```

Comment:
Since we're using the `foreach` method here for better memory performance we should not store it in that csv variable nor use map to iterate over it again.

Just using the `foreach` to iterate is enough like so:

CSV.foreach(@teams_csv, headers: true, converters: :numeric, header_converters: :symbol) do |row|
      @teams << Team.new(row.to_h)
    end

Reply:
Yes! This is a good catch. I didn't realize that at first, but still works great after pulling out csv.map!

Reply:
@guardiac @sarynm12 Nice! Just updated to `foreach` instead and removed `map`.